### PR TITLE
Various improvements to logging

### PR DESCRIPTION
**Pull request by bertfrees**
Fixes https://github.com/daisy/xproc-maven-plugin/issues/5 and some other things

Depends on:
- [x] https://github.com/daisy/xprocspec/pull/47 (included in v1.2.0)

### CODE REVIEW

**Comment by bertfrees** [comment]
@rdeltour @josteinaj Does this look good to you? I'd like this to be merged as soon as possible because the longer it takes the more time I loose while running tests.

File: xprocspec-runner/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <dependency>
       <groupId>org.daisy.xprocspec</groupId>
       <artifactId>xprocspec</artifactId>
-      <version>[1.0.0,2.0.0)</version>
+      <version>1.1.1-SNAPSHOT</version>
```
Comment (josteinaj) [inline xprocspec-runner/pom.xml]: The next version of xprocspec will probably be 1.2.0 and not 1.1.1. I'm working on it but have some issues left that I want to fix before releasing it.

**Comment by josteinaj** [comment]
xproc-engine-calabash could use an updated version of calabash. I think that might help with some issues I'm having when trying to run the xproc test suite with xprocspec. But maybe it's easier to keep xproc-engine-calabash in sync with the calabash version used in xproc-engine-daisy-pipeline.

**Comment by josteinaj** [comment]
I assume the xprocspec version referenced is the one on the master branch of xprocspec? The test_very_big takes a lot of time to run so I had to comment it out, but the rest of it builds fine.

Using 1.0.1-SNAPSHOT of xproc-maven-plugin and xproc-engine-calabash to run the tests in xprocspec works fine.

File: xprocspec-runner/pom.xml (hunk 1 above)
Comment (bertfrees) [inline xprocspec-runner/pom.xml]: OK. The things you are working on is that also logging related? If not, what about an intermediary release with just the logging stuff?

**Comment by bertfrees** [comment]
> xproc-engine-calabash could use an updated version of calabash. I think that might help with some issues I'm having when trying to run the xproc test suite with xprocspec. But maybe it's easier to keep xproc-engine-calabash in sync with the calabash version used in xproc-engine-daisy-pipeline.

The calabash versions don't have to be the same, but it is probably a bit easier planning-wise to update at the same time as we do https://github.com/daisy/pipeline-tasks/issues/85. 

> I assume the xprocspec version referenced is the one on the master branch of xprocspec?

The xprocspec version referenced is the one from daisy/xprocspec#47.

> The test_very_big takes a lot of time to run so I had to comment it out, but the rest of it builds fine.

OK

File: xprocspec-runner/pom.xml (hunk 1 above)
Comment (josteinaj) [inline xprocspec-runner/pom.xml]: It includes your log branch, yes. I'll try to get something released this week. Will work on it now.

**Comment by josteinaj** [comment]
By the way, if you want to try a newer version of xprocspec, try its dev branch. There's only a few more issues I want to have a look at before I can merge and release that branch.

File: xprocspec-runner/src/main/java/org/daisy/maven/xproc/xprocspec/XProcSpecRunner.java [hunk 2]
```diff
@@ -85,6 +88,27 @@ public boolean run(Map<String,File> tests,
 			activate();
 		engine.setCatalog(catalog);
 		
+		// register Java implementation of px:message
+		// FIXME: make a generic step that can be used by all XProc engines
+		if (engine.getClass().getName().equals("org.daisy.maven.xproc.calabash.Calabash")) {
+			try {
+				engine.getClass().getMethod("setDefaultConfiguration", Reader.class).invoke(engine, new StringReader(
```
Comment (rdeltour) [inline xprocspec-runner/src/main/java/org/daisy/maven/xproc/xprocspec/XProcSpecRunner.java]: So IIUC you set the configuration with reflection. Isn't there a risk of losing whatever is relevant in the engine's default configuration?
